pkg/kubelet/kubelet.go

```diff
@@ -1910,12 +1910,6 @@ func (kl *Kubelet) SyncTerminatingPod(_ context.Context, pod *v1.Pod, podStatus
 	klog.V(4).InfoS("SyncTerminatingPod enter", "pod", klog.KObj(pod), "podUID", pod.UID)
 	defer klog.V(4).InfoS("SyncTerminatingPod exit", "pod", klog.KObj(pod), "podUID", pod.UID)
 
-	apiPodStatus := kl.generateAPIPodStatus(pod, podStatus)
-	if podStatusFn != nil {
-		podStatusFn(&apiPodStatus)
-	}
-	kl.statusManager.SetPodStatus(pod, apiPodStatus)
-
```

Comment:
We still want this block - termination could take hours, so we want to write the status before we go into that block.

Reply:
Makes sense, it was moved down to compute the status just after termination, so that we have surely the status at hand. However, it turned out we need to delay the deletion anyway until SyncTerminatedPod, otherwise SyncTerminatedPod could operate without the cache and flip the phase - hence the recent updates.

Now we can either move this block back to its position or keep at both. WDYT?

Reply:
It should be at both

Reply:
Done